### ISSUE TITLE
[pro] bidirectional_pagination has an off by one bug?

### ISSUE THREAD

**Issue by modosc** [rmosolgo/graphql-ruby#1877]
(sorry i don't have code available, there's no specs that i could see for the pro stuff so i didn't know where to start with a reproduction.)

we've noticed an issue where paginating by N items at a time through N+1 items results in a weird error. consider we have 11 entries. 

this works as expected (
```
query { 
  entries(first: 10) { 
    pageInfo {
      hasNextPage
      hasPreviousPage
      startCursor
      endCursor
    }
  }
}
// gives us 10 entries and the following pageInfo:
    "entries": {
      "pageInfo": {
        "hasPreviousPage": false,
        "startCursor": "startCursor",
        "hasNextPage": true,
        "endCursor": "endCursor"
      },

```
and this works as expected:
```
query { 
  entries(first: 10, after: "endCursor") { 
    pageInfo {
      hasNextPage
      hasPreviousPage
      startCursor
      endCursor
    }
  }
}
// gives us 1 entry and the following pageInfo:
    "entries": {
      "pageInfo": {
        "hasPreviousPage": true,
        "startCursor": "lastCursor",
        "hasNextPage": true,
        "endCursor": "lastCursor"
      },

```
but when we go _backwards_ from here:
```
query { 
  entries(last: 10, before: "lastCursor") { 
    pageInfo {
      hasNextPage
      hasPreviousPage
      startCursor
      endCursor
    }
  }
}
// gives us our original 10 entries but `hasNextPage` is incorrect
    "entries": {
      "pageInfo": {
        "hasPreviousPage": false,
        "startCursor": "startCursor",
        "hasNextPage": false, <-- this is wrong
        "endCursor": "endCursor"
      },
```

if we do the same thing with 12 entries `hasNextPage` works as expected, hence my off-by-one suspicion.

**Comment by rmosolgo** [comment]
Ohh, interesting, so it's just `hasNextPage` showing symptoms of an off-by-one bug?  (Probably `hasPreviousPage` has the same issue if so)

**Comment by modosc** [comment]
afaict `hasPreviousPage` is working correctly, it seems to only be `hasNextPage` and only when paginating backwards.

fwiw i only found this because i added a pagination integration spec where i created N+1 items and it broke - i spent a long time debugging my own code before realizing the spec worked if i created N+2 items

**Comment by modosc** [comment]
i got excited about #2349 but it (unfortunately) doesn't solve this issue for us :/

**Comment by rmosolgo** [comment]
Hi, sorry for the very slow turn-around on this 😩 . I've just released graphql-pro 1.10.6 which should fix it! Please give it a try and let me know if you have any more trouble with it.

**Comment by modosc** [comment]
fixed, thanks! i didn't realize this was a pro issue, no wonder i couldn't track it down.
